Q:

rqt crash when subscribing to any topic

Fresh install of ubuntu gnome 16.04

apt-get install ros-kinetic-desktop-full ros-kinetic-ros-control ros-kinetic-ros-controllers ros-kinetic-moveit ros-kinetic-moveit-ros ros-kinetic-gazebo-ros

run my robot_description.launch

I run: rqt

open plugin topic monitor

check any topic

i get this:
[ERROR] [1488471809.228233]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.235319]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.242770]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.282522]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.283385]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.328485]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.336671]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.342535]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.380507]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.383168]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.428342]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.436333]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.443620]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.482359]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.483897]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.528835]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.536844]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.545590]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.587778]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.591574]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.629997]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.636860]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.644545]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.681600]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.683263]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.729279]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.736976]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.744407]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.783724]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.785033]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.829423]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.837650]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.845325]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.886005]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
[ERROR] [1488471809.890275]: bad callback: <bound method TopicInfo.message_callback of <rqt_topic.topic_info.TopicInfo object at 0x7f2ae4bc63d0>>
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
cb(msg)
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 100, in message_callback
self.sizes.append(buff.len)
AttributeError: 'cStringIO.StringO' object has no attribute 'len'
Traceback (most recent call last):
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_widget.py", line 193, in refresh_topics
self._update_topics_data()
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_widget.py", line 204, in _update_topics_data
bytes_per_s, _, _, _ = topic_info.get_bw()
File "/opt/ros/kinetic/lib/python2.7/dist-packages/rqt_topic/topic_info.py", line 120, in get_bw
max_size = max(self.sizes)
ValueError: max() arg is an empty sequence

More info:
ros-kinetic-python-qt-binding Version: 0.3.2-0xenial-20170124-171935-0800
ros-kinetic-qt-gui Version: 0.3.4-0xenial-20170124-172657-0800
ros-kinetic-rqt Version: 0.3.2-0xenial-20170124-181049-0800
python-pyqt5 Version: 5.5.1+dfsg-3ubuntu4
python-pyside2 Version: 2.0.0+dev-0~201604151742~rev1858~pkg38~ubuntu16.04.1
UPDATE:
After downloading the source from: https://github.com/ros-visualization/rqt_common_plugins
I run: catkin_make -DCMAKE_INSTALL_PREFIX=/opt/ros/kinetic install
and get:
Install the project... -- Install configuration: "" -- Installing: /opt/ros/kinetic/_setup_util.py CMake Error at cmake_install.cmake:54 (file): file INSTALL cannot copy file "/home/gortium/Software/ros-kinetic/build/catkin_generated/installspace/_setup_util.py" to "/opt/ros/kinetic/_setup_util.py".
Makefile:61: recipe for target 'install' failed make: * [install] Error 1 Invoking "make install -j4 -l4" failed

Originally posted by Gortium on ROS Answers with karma: 11 on 2017-03-02
Post score: 1

A:

Please see duplicate question #q255923.

Originally posted by Dirk Thomas with karma: 16276 on 2017-03-02
This answer was ACCEPTED on the original site
Post score: 0

Original comments
Comment by Gortium on 2017-03-02:
Thank you!
but Build fail. I updated my question.
Is it the same if i am on Kinetic?
Comment by Dirk Thomas on 2017-03-02:
Yes, it is the same.
Comment by Gortium on 2017-03-02:
Ok fixed it by not installing and sourcing the setup.bash of rqt_common_plugins
Comment by sheng on 2017-03-04:
Hello,I get the same trouble ,when run catkin_make -DCMAKE_INSTALL_PREFIX=/opt/ros/kinetic install,and get failed like you ,I can't understand what's your meaning of"not installing and sourcing the setup.bash of rqt_common_plugins",could you detailed descript it?
Comment by 130s on 2017-03-04:
@sheng see the answer suggested in this answer.
Comment by Dirk Thomas on 2017-03-04:
If you are using indigo or Jade you can simply update your installed Debian packages. The latest release of ros-<DISTRO>-rqt-topic will address the problem. The updated Kinetic package will hopefully be available on a couple of days.

